xrootd/client/client.go

```diff
@@ -44,12 +47,39 @@ type Client struct {
 	protocolVersion  int32
 	signRequirements xrdproto.SignRequirements
 	seqID            int64
+	auths            map[string]auth.Auther
+}
+
+// Option configures an XRootD client.
+type Option func(*Client) error
+
+// WithAuth adds an authentication mechanism to the XRootD client.
+// If an authentication mechanism was already registered for that provider,
+// it will be silently replaced.
+func WithAuth(a auth.Auther) Option {
+	return func(client *Client) error {
+		return client.setAuth(a)
+	}
+}
+
+func (client *Client) setAuth(auth auth.Auther) error {
```

Comment:
the way it's done now, I would rather rename `setAuth` as `addAuth`.

that said, I am not completely sure this warrants its own method, but ok. (I can live with it.) :P